x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/logic/helpers.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEqual } from 'lodash';
+
+import { Schema } from '../../../../shared/types';
+import { DEFAULT_FIELD_SETTINGS, DISABLED_FIELD_SETTINGS } from '../constants';
+import {
+  FieldResultSetting,
+  FieldResultSettingObject,
+  ServerFieldResultSetting,
+  ServerFieldResultSettingObject,
+} from '../types';
+
+const updateAllFields = (
+  fields: FieldResultSettingObject | ServerFieldResultSettingObject,
+  newValue: FieldResultSetting | {}
+) => {
+  return Object.keys(fields).reduce(
+    (acc, fieldName) => ({ ...acc, [fieldName]: { ...newValue } }),
+    {}
+  );
+};
+
+export const clearAllServerFields = (fields: ServerFieldResultSettingObject) =>
+  updateAllFields(fields, {});
+
+export const clearAllFields = (fields: FieldResultSettingObject) => updateAllFields(fields, {});
```

Comment:
[feel free to ignore if this is just a straight ent-search port] Do we need both `clearAllFields` and `clearAllServerFields` as separate fn's? theoretically could maybe just do

```js
export const clearAllFields = (fields: FieldResultSettingObject | ServerFieldResultSettingObject) => updateAllFields(fields, {});
```

and remove the need for `clearAllServerFields` 🤷 

No worries though if it's a typescript thing or if it's to make it very clear what type we're dealing with

Reply:
Ah actually I just checked, it's a straight port so I'll go ahead and collapse this